### PR TITLE
First iteration of procedures modal with block workspace

### PR DESCRIPTION
**Pull request by paulkaplan**
### Resolves

_What Github issue does this resolve (please include link)?_

Resolves https://github.com/LLK/scratch-gui/issues/831

### Proposed Changes

_Describe what this Pull Request does_

1. Adds a new reducer that follows the way the eyedropper integrates with scratch-blocks: creating a modal-ish reducer that stores a callback that is activated when the action to deactivate is called. 
2. Adds a new modal that holds a single block workspace with the procedure declaration block and buttons for modifying it.

### Test Coverage

_Please show how you have added tests to cover your changes_
Added a basic integration smoke test that creates a new custom procedure from the "Make a Block..." button, clicks all the buttons, submits the new block and checks that a "define" block has been added to the workspace. 

### Follow-up work

1. Field rendering work, which will occur in scratch-blocks
2. Design pass to cleanup styles (they are basic approximations right now, will review with @carljbowman)
2. Add the "Run without screen refresh" option (requires scratch-blocks work, so it was left out of this for now). 

add-custom-procedures-modal

### CODE REVIEW

File: src/components/custom-procedures/custom-procedures.jsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                    onClick={props.onAddTextNumber}
+                >
+                    <img
+                        className={styles.optionIcon}
+                        src={textInputIcon}
+                    />
+                    <div className={styles.optionTitle}>
+                        <FormattedMessage
+                            defaultMessage="Add an input"
+                            description="Label for button to add a numeric"
```
Comment (fsih) [inline src/components/custom-procedures/custom-procedures.jsx]: "numeric input"?

File: src/components/custom-procedures/custom-procedures.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                        <FormattedMessage
+                            defaultMessage="Add a label"
+                            description="Label for button to add a label"
+                            id="gui.customProcedures.addALabel"
+                        />
+                    </div>
+                </div>
+            </div>
+            <div className={styles.checkboxRow}>
+                {/* @todo Implement "run without screen refresh" */}
```
Comment (fsih) [inline src/components/custom-procedures/custom-procedures.jsx]: Is there a bug number we can attach?

File: src/containers/custom-procedures.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+CustomProcedures.defaultProps = {
+    options: CustomProcedures.defaultOptions
+};
+
+const mapStateToProps = state => ({
+    mutator: state.customProcedures.mutator
+});
+
+const mapDispatchToProps = () => ({});
```
Comment (fsih) [inline src/containers/custom-procedures.jsx]: Remove

**Comment by paulkaplan** [comment]
@fsih thanks! Fixed 28834b3

File: src/containers/custom-procedures.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    options: CustomProcedures.defaultOptions
+};
+
+const mapStateToProps = state => ({
+    mutator: state.customProcedures.mutator
+});
+
+export default connect(
+    mapStateToProps,
+    () => ({}) // omit dispatch prop
```
Comment (fsih) [inline src/containers/custom-procedures.jsx]: Can omit this

**Pull request merged**